### PR TITLE
GEM-144 | Add getting group attachment init from attachment store

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import pl.edu.agh.gem.integration.ability.stubInitGroupAttachment
+import pl.edu.agh.gem.internal.client.AttachmentStoreClient
+import pl.edu.agh.gem.internal.client.AttachmentStoreClientException
+import pl.edu.agh.gem.internal.client.RetryableAttachmentStoreClientException
+import pl.edu.agh.gem.util.createGroupAttachmentResponse
+
+class AttachmentStoreClientIT(
+    private val attachmentStoreClient: AttachmentStoreClient,
+) : BaseIntegrationSpec({
+    should("get available currencies") {
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt:17]: wrong name

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        stubInitGroupAttachment(attachment, userId)
+
+        // when
+        val result = attachmentStoreClient.getGroupInitAttachment(groupId, userId)
+
+        // then
+        result.id shouldBe attachment.id
+    }
+
+    should("throw CurrencyManagerClientException when we send bad request") {
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt]: wrong name

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt (hunk 1 above)
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt]: wrong name

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        val groupId = "groupId"
+        stubInitGroupAttachment(attachment, userId, NOT_ACCEPTABLE)
+
+        // when & then
+        shouldThrow<AttachmentStoreClientException> {
+            attachmentStoreClient.getGroupInitAttachment(groupId, userId)
+        }
+    }
+
+    should("throw RetryableCurrencyManagerClientException when client has internal error") {
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt]: wrong name

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt (hunk 3 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt]: fixed

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt (hunk 2 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt]: fixed

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt (hunk 1 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/client/AttachmentStoreClientIT.kt]: fixed

**Review by pwgacek — APPROVED** [review]

**Pull request merged**